Automattic/wp-calypso
# Store: Add Stripe Connect Flows

See https://github.com/Automattic/wp-calypso/issues/17002

To Test

* Test on a real Atomic site
* Verify the WooCommerce plugin is installed and activated
* Verify the WooCommerce Stripe Gateway plugin is installed and activated
* Go to the settings for the Stripe gateway and clear the publishable and secret key if present, and save your changes

* Download a ZIP of Automattic/woocommerce-services-use-staging, install and activate it on your Atomic Site
* Clone https://github.com/Automattic/woocommerce-services **(Edit: Important: Use the fix/1203-explicit-status-stripe-rest-api-errors branch)**
* `npm install`
* `npm run release`
* Install the resulting ZIP (in the release folder) and activate it on your site
* Install and activate the Stripe Connect Test Bench plugin on the site Automattic/stripe-connect-test-bench
* Go to wp-admin > WooCommerce > Status > Stripe Connect Server Testing
* In the Deauthorize section, click on Deauthorize
* It is OK if it returns an error (it might) - I just want you to be sure you're not connected

* Clone this branch (add/17002-fetch-acct-details) and `npm start` it
* Navigate to http://calypso.localhost:3000/store/{DOMAIN}
* Click on Settings and then scroll down to the Stripe section of the Payments page
* Click on Manage
* Wait while the connected status is retrieved
* Make sure you are prompted to "Take credit card payments with Stripe" and that you see a button for New Stripe account and I already have a Stripe account
* Stop

* Go to wp-admin > WooCommerce > Status > Stripe Connect Server Testing
* In the Create Account section, enter an email address you haven't used before (e.g. somename100@somewhere.com) and a valid country code (e.g. US) and hit Submit
* Verify you get an account id (e.g. acct_...)
* Navigate to http://calypso.localhost:3000/store/{DOMAIN}
* Click on Settings and then scroll down to the Stripe section of the Payments page
* Click on Manage
* Wait while the connected status is retrieved
* Make sure you are presented with the email address you chose and a message prompting you to check your email to activate your account
* Note - the disconnect button is not hooked up yet

* Lastly, run the new unit tests with

```
npm run test-client client/extensions/woocommerce/state/
```
